src/SwapRouter.sol

```diff
@@ -0,0 +1,210 @@
+        IERC20Metadata token1 = direction ? IERC20Metadata(token) : USDC;
+
+        token0.safeTransferFrom(recipient, address(this), amountToSwap);
+
+        uint256 expectedAmountOut = getTokenPriceInUSD(address(token1));
+
+        _swapTokens(
+            token0,
+            token1,
+            amountToSwap,
+            expectedAmountOut,
+            slippage,
+            data
+        );
+        amountOut = token1.balanceOf(address(this));
+
+        token1.safeTransfer(recipient, amountOut);
+        token0.safeTransfer(recipient, token0.balanceOf(address(this)));
+    }
+
+    /// @dev Direction => True => CRV -> CVXCRV
+    /// @dev Direction => False =>  CVXCRV -> CRV
+    function swapOnCRVCVXCRVPool(
+        bool direction,
+        uint256 amount,
+        address recipient
+    ) external override returns (uint256 amountOut) {
+        IERC20Metadata swapToken = direction ? CRV : CVXCRV;
+        IERC20Metadata recievedToken = direction ? CVXCRV : CRV;
+
+        require(
+            amount > 0 && amount <= swapToken.balanceOf(msg.sender),
+            "SwapRouter :: amount"
+        );
+        require(recipient != address(0), "SwapRouter :: recipient");
+
+        swapToken.safeTransferFrom(msg.sender, address(this), amount);
+        crvcvxcrvPool.exchange(
+            int8(direction ? 0 : 1),
+            int8(direction ? 1 : 0),
+            amount,
+            0,
+            address(this)
+        );
+
+        amountOut = recievedToken.balanceOf(address(this));
+
+        swapToken.safeTransfer(recipient, swapToken.balanceOf(address(this)));
+        recievedToken.safeTransfer(recipient, amountOut);
+    }
+
+    function burn3CRVForUSDC(uint256 amount, address recipient)
+        external
+        override
+        returns (uint256 amountOut)
+    {
+        require(
+            amount > 0 && amount <= _3CRV.balanceOf(msg.sender),
+            "SwapRouter :: amount"
+        );
+        require(recipient != address(0), "SwapRouter :: recipient");
+
+        _3CRV.safeTransferFrom(msg.sender, address(this), amount);
+        /// @dev i = 1 --> USDC
+        _3crvPool.remove_liquidity_one_coin(amount, 1, 0);
+
+        amountOut = USDC.balanceOf(address(this));
+
+        _3CRV.safeTransfer(recipient, amount);
+        USDC.safeTransfer(recipient, amountOut);
+    }
+
+    function getTokenPriceInUSD(address token) public view returns (uint256) {
+        require(
+            token == address(CRV) || token == address(CVX),
+            "SwapRouter :: token"
+        );
+
+        (, int256 answer, , , ) = (token == address(CRV) ? CRVUSD : CVXUSD)
+            .latestRoundData();
+
+        return (uint256(answer) / uint256(CRVUSD.decimals())) * USDC.decimals();
+    }
+
+    function _swapTokens(
+        IERC20Metadata token0,
+        IERC20Metadata token1,
+        uint256 amount,
+        uint256 minReturn,
+        uint256 slippage,
+        bytes memory data
```

Comment:
Decode data and add checks for tokens and amounts before initiating swap

Reply:
not needed